### PR TITLE
Feature: Add authentication using leto-modelizer-api

### LINKED ISSUE #425: Add the authentication on the front
## Description

**As a** `user` ,

**I want** to be able to log myself using the backend (through github)

To do so, we want a button that will redirect us on the correct provider (here github) and the use the provider to connect to the application using the backend.
Moreover, what previosly has been done using OIDC in the front, should be removed.
Also the button of auth will be done according to the Catherine's mockup

### PR DESCRIPTION
**Pull request by aboubakar14**
resolve #425

### CODE REVIEW

File: tests/unit/stores/UserStore.spec.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+describe('User Store', () => {
+  beforeEach(() => {
+    // creates a fresh pinia and makes it active
+    // so it's automatically picked up by any useStore() call
+    // without having to pass it to it: `useStore(pinia)`
+    setActivePinia(createPinia());
+  });
+
+  it('should test variables in the store', () => {
```
Comment (Zorin95670) [inline tests/unit/stores/UserStore.spec.js:12]: missing test username empty and firstname not

Comment (Zorin95670) [reply tests/unit/stores/UserStore.spec.js:12]: And separate test into multiple it like
describe('Test getter: isEmpty')
with it(should return true on empty username and firstname) etc etc

File: quasar.config.js [hunk 2]
```diff
@@ -108,9 +108,7 @@ module.exports = configure((ctx) => ({
     env: {
       TEMPLATE_LIBRARY_BASE_URL: process.env.TEMPLATE_LIBRARY_BASE_URL || configuration?.templateLibrary || '',
       VERSION: version,
-      AUTHENTICATION: configuration?.authentication?.OIDC
-        ? JSON.stringify(configuration.authentication.OIDC)
-        : '',
+      IFBACKEND: configuration?.backend ? Object.prototype.hasOwnProperty.call(configuration?.backend, 'url') && Object.prototype.hasOwnProperty.call(configuration?.backend, 'appId') : false,
```
Comment (Zorin95670) [inline quasar.config.js:111]: IFBACKEND ???
may be HAS_BACKEND or something better

File: quasar.config.js [hunk 3]
```diff
@@ -157,6 +155,12 @@ module.exports = configure((ctx) => ({
         secure: true,
         changeOrigin: true,
       },
+      '/backend': {
```
Comment (Zorin95670) [inline quasar.config.js:158]: rename to /api

File: src/composables/BackendApi.js [hunk 4]
```diff
@@ -0,0 +1,59 @@
+import axios from 'axios';
+
+const configuration = require('../../global.config.json');
```
Comment (Zorin95670) [inline src/composables/BackendApi.js:3]: 0_0
why not import ?

File: src/composables/BackendApi.js [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+}
+
+/**
+ * Retrieve (GET request) information about the current user.
+ * @param {string} sessionToken The current user's session token.
+ * @returns {Promise<object>} Promise with the logged-in user information on success
+ * otherwise the error that happened during the request.
+ */
+export async function getUserInfo(sessionToken) {
+  const config = {
```
Comment (Zorin95670) [inline src/composables/BackendApi.js:51]: do it in .get directly

File: src/composables/UserAuthentication.js [hunk 6]
```diff
@@ -1,6 +1,8 @@
 import { UserManager, WebStorageStateStore } from 'oidc-client-ts';
 import nunjucks from 'nunjucks';
 import User from 'src/models/User';
+import * as api from 'src/composables/BackendApi';
```
Comment (Zorin95670) [inline src/composables/UserAuthentication.js:1]: import { login, getUserInfo } from

File: src/composables/UserAuthentication.js [hunk 7]
```diff
@@ -46,14 +48,64 @@ export function setUserManager(providerName) {
...
+export function getUserSessionToken() {
+  return localStorage.getItem('sessionToken');
+}
+
 /**
  * Initiate the login process by redirecting the user to the authentication provider.
  * @returns {Promise<void>} Promise with nothing on success otherwise an error.
  */
-export async function login() {
+export async function logine() {
```
Comment (Zorin95670) [inline src/composables/UserAuthentication.js]: logine ???

Comment (Zorin95670) [reply src/composables/UserAuthentication.js]: think is useless code

File: src/composables/UserAuthentication.js [hunk 8]
```diff
@@ -46,14 +48,64 @@ export function setUserManager(providerName) {
...
   return userManager.signinRedirect();
 }
 
+/**
+ * Log in the user using its temporaryCode. Once the user is logged, it saves
+ * the user information in the dedicated store.
+ * @param {string} temporaryCode The temporary code obtained from the github provider.
```
Comment (Zorin95670) [inline src/composables/UserAuthentication.js:23]: temporaryCode - The

File: src/composables/UserAuthentication.js [hunk 9]
```diff
@@ -46,14 +48,64 @@ export function setUserManager(providerName) {
...
+ * @returns {Promise<object>} Promise with the logged-in user information on success
+ * otherwise the error that happened during the request.
+ */
+export async function login(temporaryCode) {
+  return api.login(temporaryCode)
+    .then((response) => {
+      const userStore = useUserStore();
```
Comment (Zorin95670) [inline src/composables/UserAuthentication.js:30]: break line after

File: src/composables/UserAuthentication.js [hunk 10]
```diff
@@ -46,14 +48,64 @@ export function setUserManager(providerName) {
...
+      setUserSessionToken(response.data.sessionToken);
```
Comment (Zorin95670) [inline src/composables/UserAuthentication.js:31]: break line after

File: src/pages/SplashPage.vue [hunk 11]
```diff
@@ -26,15 +26,62 @@
...
 const router = useRouter();
 
+/**
+ * Initialize the user for the appplication, if needed.
+ * If the backend is activated in the configuration, it will check
+ * that the session token is present or not. If not, it will try to login the user.
+ * If already connected, it will just retrieve the user info.
+ */
+async function initUser() {
+  if (process.env.IFBACKEND) {
```
Comment (Zorin95670) [inline src/pages/SplashPage.vue:49]: wow non
prefere:
```js
if (!process.env.HAS_BACKEND) {
  return;
}
```

File: src/pages/SplashPage.vue [hunk 12]
```diff
@@ -26,15 +26,62 @@
...
+    // TODO: Add another if to check if the session token is expired
+    if (!getUserSessionToken()) {
```
Comment (Zorin95670) [inline src/pages/SplashPage.vue:51]: same:

if (!getUserSettionToken) {
  return login(...)...;
}

File: src/pages/SplashPage.vue [hunk 13]
```diff
@@ -26,15 +26,62 @@
...
+          });
+        })
+        .catch(() => {
+          Notify.create({
+            type: 'negative',
+            message: t('errors.authentication.login'),
+            html: true,
+          });
+        });
+    } else if (getUserSessionToken() && useUserStore().isEmpty) {
```
Comment (Zorin95670) [inline src/pages/SplashPage.vue:67]: same as before

File: src/router/index.js [hunk 14]
```diff
@@ -36,20 +37,27 @@ export default route((/* { store, ssrContext } */) => {
...
+
+    if (!authRoutes.includes(to.name) && !isUserReady && process.env.IFBACKEND) {
+      window.location.href = backendUrl.data;
     } else if (isUserReady
       && !applicationReady
       && to.name !== 'Splash') {
-      next({ name: 'Splash', query: { from: to.fullPath } });
+      let temporaryCode;
+      if (isComingFromGithub) {
+        temporaryCode = window.location.href.match(/\?code=(.*)/)[1].substring(0, 20);
```
Comment (Zorin95670) [inline src/router/index.js:57]: replace .* by [^&]*

File: src/router/index.js [hunk 15]
```diff
@@ -44,11 +44,10 @@ export default route(async (/* { store, ssrContext } */) => {
   }
 
   Router.beforeEach(async (to, from, next) => {
-    const authRoutes = ['SilentRefresh', 'Redirect', 'Login'];
     const isComingFromGithub = window.location.href.search(/\?code=(.*)/) !== -1;
     const isUserReady = getUserSessionToken() || isComingFromGithub || !process.env.IFBACKEND;
 
-    if (!authRoutes.includes(to.name) && !isUserReady && process.env.IFBACKEND) {
+    if (!isUserReady && process.env.IFBACKEND) {
```
Comment (Zorin95670) [inline src/router/index.js:50]: put all of this file modification is the commit you update the router.

File: README.md [hunk 16]
```diff
@@ -161,101 +161,26 @@ When you have installed all the desired plugins, please run this commands `npm r
 ![](docs/plugin-init.png)
 
 
-## How to setup the authentication with OIDC
+## How to setup the authentication with Leto-Modelizer-Api
 
-We use OIDC authentication with [oidc-client-ts](https://github.com/authts/oidc-client-ts) library. To do so, you must fill the configuration related to the provider of your choice inside `global.config.json` root file. You can declare one or more provider for the user to log in.
-Example with `Google` and `Github` providers.
+We are using [Leto-Modelizer-Api](https://github.com/ditrit/leto-modelizer-api) authentication (as a backend). To setup it, you must fill the configuration inside `global.config.json` root file.
```
Comment (Zorin95670) [inline README.md:166]: Missing sentence to explain that a admin front is coming.

File: quasar.config.js (hunk 3 above)
Comment (aboubakar14) [reply quasar.config.js:158]: Saw with @Zorin95670

File: src/composables/UserAuthentication.js (hunk 6 above)
Comment (aboubakar14) [reply src/composables/UserAuthentication.js:1]: Saw with @Zorin95670

File: src/pages/SplashPage.vue (hunk 12 above)
Comment (aboubakar14) [reply src/pages/SplashPage.vue:51]: I do not agree with you on that one. Because it makes the code less readable.

Moreover I can't really separate these 2 if because they are already 2 separate condition in which case we do not have the same behavior.

File: tests/unit/stores/UserStore.spec.js (hunk 1 above)
Comment (aboubakar14) [reply tests/unit/stores/UserStore.spec.js:12]: Done

File: quasar.config.js (hunk 2 above)
Comment (aboubakar14) [reply quasar.config.js]: done

File: src/composables/BackendApi.js (hunk 4 above)
Comment (aboubakar14) [reply src/composables/BackendApi.js]: done

File: src/composables/BackendApi.js (hunk 5 above)
Comment (aboubakar14) [reply src/composables/BackendApi.js]: done

File: src/composables/UserAuthentication.js (hunk 7 above)
Comment (aboubakar14) [reply src/composables/UserAuthentication.js]: already deleted in the correct commit

File: src/composables/UserAuthentication.js (hunk 8 above)
Comment (aboubakar14) [reply src/composables/UserAuthentication.js]: done

File: src/composables/UserAuthentication.js (hunk 9 above)
Comment (aboubakar14) [reply src/composables/UserAuthentication.js:30]: done

File: src/composables/UserAuthentication.js (hunk 10 above)
Comment (aboubakar14) [reply src/composables/UserAuthentication.js:32]: done

File: src/pages/SplashPage.vue (hunk 11 above)
Comment (aboubakar14) [reply src/pages/SplashPage.vue]: done

File: src/router/index.js (hunk 14 above)
Comment (aboubakar14) [reply src/router/index.js]: done

File: src/plugins/index.js [hunk 17]
```diff
@@ -1,3 +1,5 @@
-// Currently empty but should be filled by CLI (cf. README)
+import terratorPlugin from 'terrator-plugin';
 
-export default {};
+export default {
+terratorPlugin,
```
[the remark this answers is not in the document]
Comment (aboubakar14) [reply src/plugins/index.js]: removed

File: src/router/index.js (hunk 15 above)
Comment (aboubakar14) [reply src/router/index.js]: done

File: tests/unit/stores/UserStore.spec.js [hunk 18]
```diff
@@ -0,0 +1,43 @@
...
+
+  describe('test store actions', () => {
+    describe('test isEmpty', () => {
+      it('should return true if username and firstname are empty', () => {
+        const store = useUserStore();
+
+        expect(store.isEmpty).toBeTruthy();
+      });
+
+      it('should return false if username and firstname are empty', () => {
```
Comment (justinehell) [inline tests/unit/stores/UserStore.spec.js:33]: *are not empty or are defined

File: quasar.config.js [hunk 19]
```diff
@@ -157,6 +156,12 @@ module.exports = configure((ctx) => ({
...
+        pathRewrite: { '^/backend': '' },
+        target: configuration?.backend?.url,
+        changeOrigin: true,
+
```
Comment (justinehell) [inline quasar.config.js:163]: remove empty line

**Review by justinehell — CHANGES_REQUESTED** [review]

File: src/composables/LetoModelizerApi.js [hunk 20]
```diff
@@ -0,0 +1,61 @@
...
+ * otherwise the error that happened during the request.
+ */
+export async function getAuthenticationUrl() {
+  return axios.get('backend/authenticationUrl');
+}
+
+/**
+ * Log the user through the Parse-Server backend using the temporary code
+ * obtained from the github provider (from the url given by the backend).
+ * @param {string} temporaryCode The temporary code obtained from the github provider.
```
Comment (justinehell) [inline src/composables/LetoModelizerApi.js:15]: temporaryCode - The temporary code

File: src/composables/LetoModelizerApi.js [hunk 21]
```diff
@@ -0,0 +1,61 @@
...
+  };
+
+  return axios.post('/backend/api/functions/GitHubAuth', params, headers)
+    .then((response) => {
+      const firstName = response.data.result.firstname;
+      const userName = response.data.result.username;
+      const userParams = {
+        authData: response.data.result.authData,
+        firstname: firstName,
+        username: userName,
```
Comment (justinehell) [inline src/composables/LetoModelizerApi.js:37]: could not juste write like this?
```js
const userParams = {
   ...,
   firstname,
   username,
}
```

File: src/composables/LetoModelizerApi.js [hunk 22]
```diff
@@ -0,0 +1,61 @@
...
+      };
+
+      // This call is to link the user to its provider (github).
+      // In Parse-Server this call must be done using the previous received result (i.e authData).
```
Comment (justinehell) [inline src/composables/LetoModelizerApi.js:41]: it's authData in the doc, but you passed the whole userParams object, why?

File: src/composables/LetoModelizerApi.js [hunk 23]
```diff
@@ -0,0 +1,61 @@
...
+      return axios.post('/backend/api/users/', userParams, headers);
+    });
+}
+
+/**
+ * Retrieve (GET request) information about the current user.
+ * @param {string} sessionToken The current user's session token.
```
Comment (justinehell) [inline src/composables/LetoModelizerApi.js:48]: sessionToken - The current ...

File: tests/unit/composables/LetoModelizerApi.spec.js [hunk 24]
```diff
@@ -0,0 +1,104 @@
...
+            read: true,
+            write: true,
+          },
+        },
+        sessionToken: 'r:dead779dcda4970cc7f96c09a328d771',
+      };
+      axios.get.mockResolvedValueOnce(resultPostUser);
+
+      const res = await getUserInfo();
+      expect(res.sessionToken).toEqual('r:dead779dcda4970cc7f96c09a328d771');
```
Comment (justinehell) [inline tests/unit/composables/LetoModelizerApi.spec.js:101]: we only retrieve the sessionToken, not the username and the firstname ?

File: src/composables/UserAuthentication.js [hunk 25]
```diff
@@ -46,14 +48,67 @@ export function setUserManager(providerName) {
   });
 }
 
+/**
+ * Set the current user's session token in the local storage.
+ * @param {string} sessionToken The session token to store.
```
Comment (justinehell) [inline src/composables/UserAuthentication.js:6]: sessionToken - The session token...

File: tests/unit/composables/UserAuthentication.spec.js [hunk 26]
```diff
@@ -115,14 +199,34 @@ describe('User Authentication', () => {
   });
 
   describe('Test function: login', () => {
-    it('should call signinRedirect on userManager', async () => {
-      process.env.AUTHENTICATION = JSON.stringify(authentication);
+    it('should login using the api and store the user info in pina', async () => {
```
Comment (justinehell) [inline tests/unit/composables/UserAuthentication.spec.js:91]: pinia

File: tests/unit/pages/SplashPage.spec.js [hunk 27]
```diff
@@ -63,4 +78,54 @@ describe('Test component: SplashPage', () => {
       expect(setTimeout).toHaveBeenLastCalledWith(expect.any(Function), 2000);
     });
   });
+
+  describe('Test function: initUser', () => {
+    it('should call backend is activated and session token is not in storage', async () => {
```
Comment (justinehell) [inline tests/unit/pages/SplashPage.spec.js:83]: `should call backend is activated` -> missing word(s) ?

File: tests/unit/pages/SplashPage.spec.js [hunk 28]
```diff
@@ -63,4 +78,54 @@ describe('Test component: SplashPage', () => {
...
+      getUserSessionToken.mockImplementation(() => false);
+      login.mockImplementation(() => Promise.resolve({}));
+      Notify.create = jest.fn();
+
+      await wrapper.vm.initUser();
+      expect(login).toHaveBeenCalled();
+      expect(Notify.create).toHaveBeenCalledWith(expect.objectContaining({ type: 'positive' }));
+    });
+
+    it('should call backend is activated but login failed', async () => {
```
Comment (justinehell) [inline tests/unit/pages/SplashPage.spec.js:95]: same for the following tests, missing word

File: src/composables/UserAuthentication.js [hunk 29]
```diff
@@ -1,90 +1,55 @@
...
+ * otherwise the error that happened during the request.
  */
-export async function getUser() {
-  return userManager?.getUser().then((user) => (user
-    ? new User({
-      firstname: nunjucks.renderString(`{{ ${userAuthentication?.userMapping?.firstname} }}`, user),
-      lastname: nunjucks.renderString(`{{ ${userAuthentication?.userMapping?.lastname} }}`, user),
-      email: nunjucks.renderString(`{{ ${userAuthentication?.userMapping?.email} }}`, user),
-      id: nunjucks.renderString(`{{ ${userAuthentication?.userMapping?.id} }}`, user),
+export async function getUserInfo(sessionToken) {
```
Comment (Zorin95670) [inline src/composables/UserAuthentication.js:46]: you don't return userInformation at the end of .then

Comment (Zorin95670) [reply src/composables/UserAuthentication.js:46]: so change name by initUserInformations or return userInformation. But I prefer first solution.

File: README.md [hunk 30]
```diff
@@ -161,101 +161,27 @@ When you have installed all the desired plugins, please run this commands `npm r
...
```
Comment (justinehell) [inline README.md:166]: `setup it` sounds weird, I would write `to set it up` or `to configure it`

**Review by justinehell — CHANGES_REQUESTED** [review]

File: src/composables/LetoModelizerApi.js (hunk 21 above)
Comment (justinehell) [inline src/composables/LetoModelizerApi.js:37]: why create variable for firstName and userName but not authData?

**Review by justinehell — CHANGES_REQUESTED** [review]

File: src/composables/UserAuthentication.js [hunk 31]
```diff
@@ -1,90 +1,55 @@
...
+  return api.getUserInfo(sessionToken)
+    .then((response) => {
+      const userStore = useUserStore();
+
+      userStore.username = response.data.username;
+      userStore.firstname = response.data.firstname;
     })
-    : null));
+    .catch(() => { throw new Error('Could not retrieve user data'); });
```
Comment (Zorin95670) [inline src/composables/UserAuthentication.js:54]: no need to overide catch.

File: src/composables/LetoModelizerApi.js (hunk 21 above)
Comment (aboubakar14) [reply src/composables/LetoModelizerApi.js]: It was some remaning code from when I was debugging, thanks ;)

**Review by justinehell — APPROVED** [review]

File: src/composables/UserAuthentication.js (hunk 7 above)
Comment (Zorin95670) [reply src/composables/UserAuthentication.js]: You can't leave this in commit. You have to delete this function in this commit (Add new way of authenticate the user). In doing that, github will show us that you have modified the content of function and its documenation. That would be better.

File: nginx.conf [hunk 32]
```diff
@@ -20,14 +20,6 @@ http {
       proxy_pass "https://$uri$is_args$args";
     }
 
-    ## Add this for each authentication provider url defined inside configuration file.
-    # location "/PROVIDER_AUTH/" {
```
Comment (Zorin95670) [inline nginx.conf:24]: you have to keep this comment but update it for new backend api

File: src/composables/LetoModelizerApi.js [hunk 33]
```diff
@@ -0,0 +1,59 @@
...
+    });
+}
+
+/**
+ * Retrieve (GET request) information about the current user.
+ * @param {string} sessionToken - The current user's session token.
+ * @returns {Promise<object>} Promise with the logged-in user information on success
+ * otherwise the error that happened during the request.
+ */
+export async function getUserInfo(sessionToken) {
```
Comment (Zorin95670) [inline src/composables/LetoModelizerApi.js:50]: getUserInformation

File: src/composables/LetoModelizerApi.js [hunk 34]
```diff
@@ -0,0 +1,59 @@
...
+  const headers = {
+    headers: {
+      Accept: 'application/json',
+      'X-Parse-Application-Id': process.env.BACKEND_APP_ID,
+      'X-Parse-Session-Token': sessionToken,
+    },
+  };
```
Comment (Zorin95670) [inline src/composables/LetoModelizerApi.js:57]: break line after.

File: src/composables/LetoModelizerApi.js [hunk 35]
```diff
@@ -0,0 +1,59 @@
...
+ * @returns {Promise<object>} Promise with the logged-in user information on success
+ * otherwise the error that happened during the request.
+ */
+export async function login(temporaryCode) {
+  const headers = {
+    headers: {
+      Accept: 'application/json',
+      'X-Parse-Application-Id': process.env.BACKEND_APP_ID,
+    },
+  };
```
Comment (Zorin95670) [inline src/composables/LetoModelizerApi.js:25]: break line after.

**Review by Zorin95670 — APPROVED** [review]

File: src/stores/store-flag.d.ts [hunk 36]
```diff
@@ -0,0 +1,10 @@
+/* eslint-disable */
+// THIS FEATURE-FLAG FILE IS AUTOGENERATED,
+//  REMOVAL OR CHANGES WILL CAUSE RELATED TYPES TO STOP WORKING
```
Comment (k-y0u) [inline src/stores/store-flag.d.ts:3]: maybe a space too much before `REMOVAL`

Comment (k-y0u) [reply src/stores/store-flag.d.ts:3]: and probably a dot at the end of sentence too

File: src/stores/store-flag.d.ts [hunk 37]
```diff
@@ -0,0 +1,10 @@
...
+import "quasar/dist/types/feature-flag";
+
+declare module "quasar/dist/types/feature-flag" {
+  interface QuasarFeatureFlags {
+    store: true;
```
Comment (k-y0u) [inline src/stores/store-flag.d.ts:8]: should it be a coma instead of a semi-column ?

File: README.md [hunk 38]
```diff
@@ -305,13 +231,27 @@ http {
 }
 ```
 
-**_NOTE:_**  You can use the global configuration file `global.config.json` to define `TEMPLATE_LIBRARY_BASE_URL` environment variable like so :
+Same your Leto-Modelizer-Api, you need to set a reverse proxy named `backend`. You can see an example below of an `nginx.conf` file:
```
Comment (justinehell) [inline README.md:234]: Same for your Leto...

File: tests/unit/stores/UserStore.spec.js [hunk 39]
```diff
@@ -0,0 +1,43 @@
+import { setActivePinia, createPinia } from 'pinia';
+import { useUserStore } from 'src/stores/UserStore';
+
+describe('User Store', () => {
+  beforeEach(() => {
+    // creates a fresh pinia and makes it active
+    // so it's automatically picked up by any useStore() call
+    // without having to pass it to it: `useStore(pinia)`
```
Comment (k-y0u) [inline tests/unit/stores/UserStore.spec.js:8]: `to pass it to it` not clear

**Review by justinehell — CHANGES_REQUESTED** [review]

File: tests/unit/stores/UserStore.spec.js [hunk 40]
```diff
@@ -0,0 +1,43 @@
...
+    setActivePinia(createPinia());
+  });
+
+  it('should test variables in the store', () => {
```
Comment (k-y0u) [inline tests/unit/stores/UserStore.spec.js:12]: describe('Test store variable: username')
it('should be initialized to an empty string')
it('should return the new value after being set')

describe('Test store variable: username')
it('should be initialized to an empty string')
it('should return the new value after being set')

File: tests/unit/stores/UserStore.spec.js [hunk 41]
```diff
@@ -0,0 +1,43 @@
...
+
+    store.username = 'username';
+    store.firstname = 'firstname';
+
+    expect(store.username).toBe('username');
+    expect(store.firstname).toBe('firstname');
+  });
+
+  describe('test store actions', () => {
+    describe('test isEmpty', () => {
```
Comment (k-y0u) [inline tests/unit/stores/UserStore.spec.js:26]: describe('Test store action: isEmpty')

File: tests/unit/stores/UserStore.spec.js [hunk 42]
```diff
@@ -0,0 +1,43 @@
...
+      it('should return true if username and firstname are empty', () => {
+        const store = useUserStore();
```
Comment (k-y0u) [inline tests/unit/stores/UserStore.spec.js:28]: could be externalized outside of the itS

**Review by k-y0u — CHANGES_REQUESTED** [review]

File: tests/unit/stores/UserStore.spec.js (hunk 42 above)
Comment (aboubakar14) [reply tests/unit/stores/UserStore.spec.js:28]: Tried it. Unfortunately I can't because we need to set the pinia store in each test using setActivePinia (which is done in the beforeEach)

File: README.md [hunk 43]
```diff
@@ -161,101 +161,27 @@ When you have installed all the desired plugins, please run this commands `npm r
...
-
-  `metadata`: Additional metadata related to the OIDC provider configuration. It includes properties like authorization_endpoint, token_endpoint, and others.
-
-  `authorization_endpoint`: URL of the OP's OAuth 2.0 Authorization Endpoint.
-
-  `token_endpoint`: The URL of the OIDC provider's token endpoint. It is used to exchange the authorization code for an access token.
-
-:warning: Don't forget to update `nginx.conf` with a reverse proxy for your authentication provider.
+**_NOTE:_**: If the previous configuration is not present in the configuration file, Leto-Modelizer will be launched with the backend mode deactivated.
+**_NOTE:_**: For now, there is no UI associated to the backend, but the UI for the admin is coming soon !
```
Comment (k-y0u) [inline README.md:184]: waiting...

File: changelog.md [hunk 44]
```diff
@@ -25,6 +25,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
   * Work with [terrator-plugin](https://github.com/ditrit/terrator-plugin/tree/0.7.1) version 0.7.1.
   * Work with [githubator-plugin](https://github.com/ditrit/githubator-plugin/tree/0.3.0) version 0.3.0.
 * Improve performance on git add.
+* Replace OIDC authentication by Leto-Modelizer-Api, see [this issue](https://github.com/ditrit/leto-modelizer/issues/425).
```
Comment (k-y0u) [inline changelog.md:28]: should be the last commit

**Review by k-y0u — CHANGES_REQUESTED** [review]

File: src/pages/SplashPage.vue [hunk 45]
```diff
@@ -26,15 +26,63 @@
...
+ * Initialize the user for the appplication, if needed.
+ * If the backend is activated in the configuration, it will check
+ * that the session token is present or not. If not, it will try to login the user.
+ * If already connected, it will just retrieve the user info.
+ */
+async function initUser() {
+  if (!process.env.HAS_BACKEND) {
+    return;
+  }
+  // TODO: Add another if to check if the session token is expired
```
[the remark this answers is not in the document]
Comment (aboubakar14) [reply src/pages/SplashPage.vue:52]: Nop

File: tests/unit/components/menu/ModelizerSettingsMenu.spec.js [hunk 46]
```diff
@@ -72,8 +75,8 @@ describe('Test component: ModelizerSettingsMenu', () => {
     });
 
     describe('Test ref: user', () => {
-      it('should match an empty object', () => {
-        expect(wrapper.vm.user).toEqual({});
+      it('should match a pinia object', () => {
+        expect(wrapper.vm.userStore).not.toEqual({});
```
[the remark this answers is not in the document]
Comment (aboubakar14) [reply tests/unit/components/menu/ModelizerSettingsMenu.spec.js:79]: Did not find another way because the object is really big. So my thought was to just check that it was not an empty object !
But I will still add a comment to clarify the test

**Review by k-y0u — APPROVED** [review]

File: CONTRIBUTING.md [hunk 47]
```diff
@@ -149,61 +149,4 @@ Release tag are only done on the `main` branch.
     * `prerelease`, used for preparing the branch for the release
     * `release`, used for releasing project
     * `hotfix`, used for applying a hotfix on main
-* `BRANCH_NAME` is managed by this regex: `[a-z0-9._-]` (`_` is used as space character).
```
Comment (Zorin95670) [inline CONTRIBUTING.md:152]: bad deletion

**Review by Zorin95670 — APPROVED** [review]

**Review by justinehell — APPROVED** [review]
